### DISCUSSION TITLE
Welcome to OSPP 2023 Discussions!

### DISCUSSION THREAD

**Discussion by faweizhao26** [kubesphere/community#427] [General]
👋 Welcome!
We’re using Discussions as a place to connect with other members of SOPP 2023. We hope that you:

Ask questions you’re wondering about the project of OSPP 2023.

- Show your progress of your project.
- Share ideas.
- Engage with other members.

So you can comment below with your questions or progress or ideas.

**Comment by yi-ge-dian** [comment]
# OSPP 2023 - Nodejs ff dev
## Project: [OpenFunction's Node.js function framework supports Dapr state management](https://summer-ospp.ac.cn/org/prodetail/236690197?list=org&navpage=org)
## Area: Functions-as-a-Service
## Reporter: Wenlong Dong

### WWeekly Meeting 1st Report
- **prerequisite**: 
  - typescritpt learning
  - javascript learning
  - node js framework test running
- **Progress**: [Feature1-support state store](https://github.com/OpenFunction/functions-framework-nodejs/pull/189) reviewed twice, and almost be finished
- **Problem**: There are still some [problems](https://docs.dapr.io/developing-applications/building-blocks/state-management/howto-state-query-api/) with queries using the redis database, I will take a closer look.

### WWeekly Meeting 2nd Report
- **Progress**: [Feature1-support state store](https://github.com/OpenFunction/functions-framework-nodejs/pull/189) finished
- **Progress**: [Feature2-add e2e test](https://github.com/OpenFunction/functions-framework-nodejs/pull/200) doing
  - [x] learn [skywalking-infra-e2e](https://skywalking.apache.org/docs/skywalking-infra-e2e/next/readme/), [go ff](https://github.com/OpenFunction/functions-framework-go) and [kind](https://kind.sigs.k8s.io/)
  - [x] Sync-http
  - [x] Sync-cloudEvent
  - [ ] Bindings
  - [ ] PubSubs
  - [ ] Plugins
  - [ ] Statestore

### WWeekly Meeting 3rd Report
- **Progress**: [Feature2-add e2e test](https://github.com/OpenFunction/functions-framework-nodejs/pull/200) doing
- [x] refactor the test
  - [x] Knative-http
  - [x] Knative-cloudevent
  - [x]  Knative-event
  - [x] Knative-openfunction-bindings
  - [x] Knative-openfunction-pubsub
  - [ ] Knative-openfunction-statestore
  - [x] Async-openfunction-bindings
  - [ ] Async-openfunction-pubSub
  - [ ] Plugins

### WWeekly Meeting 4th Report
- finish all
- wait to pr

### WWeekly Meeting 5th Report
- finish all
- sharing experience in the  wweekly meeting

**Comment by shijilin0116** [comment]
### Weekly Meeing Report
- Project：KubeKey

- Reporter: Jilin Shi

- Progress: 
  - Read the source code of kubekey.
  - Added the “kk console” command for controlling the status of the web console, including start, shutdown and status.
  - Implemented starting and shutting down the HTTP server using the Gin framework in the “kk console start” and “kk console shutdown” commands.
  - Learning React and KubeDesign.

**Reply by shijilin0116** [reply]
### Weekly Meeing Report 2
- Project：KubeKey
- Reporter: Jilin Shi
- Progress:
  - Filled different functions into various routes of the gin framework
  - Continue learning React and building frontend pages
- Plan:
  - Implement websocket for transmitting logs to the frontend
  - continue writing frontend code

**Reply by shijilin0116** [reply]
### Weekly Meeing Report 3
- Project：KubeKey
- Reporter: Jilin Shi
- Progress:
  - Implemented websocket for transmitting logs to the frontend
- Plan:
  - Organize existing code
  - continue writing frontend code

**Reply by shijilin0116** [reply]
### Weekly Meeing Report 4
- Project：KubeKey
- Reporter: Jilin Shi
- Progress:
  - Complete the static content code according to the prototype.
  - Implement partial dynamic content code.
- Plan:
  - complete dynamic content code.
  - Submit the completed portion as a PR.

**Reply by shijilin0116** [reply]
### Weekly Meeing Report 5
- Project：KubeKey
- Reporter: Jilin Shi
- Progress:
  - Complete dynamic content code.
  - Reorganize the structure of the backend Gin routes.
  - Adjust the layout.
  - Optimize rendering performance.
- Plan:
  - Enhance interactivity for the displayed log section.
  - Add support for artifact functionality.
  - UI test and unit test.
  - Write documentation.

**Reply by shijilin0116** [reply]
### Weekly Meeing Report 6
- Project：KubeKey
- Reporter: Jilin Shi
- Progress:
  - Finish all code.
- Plan:
  - Package static resources into the kk program.
  - Submit codes.
  - Write documentation.

**Comment by Ganbingkun** [comment]
# weekly report
- Project: KubeSphere-S2I
- Reporter: Bingkun Gan
- Progress: 
   - Further clarify requirements
   - Read S2I source code
   - Learning kubebuilder buildpacks shipwright

**Reply by Ganbingkun** [reply]
# weekly report 2
- Project: KubeSphere-S2I
- Reporter: Bingkun Gan
- Progress: 
   - Created base images based on multiple languages and versions
   - Implemented Shipwright Demo
- Planning: 
   - Discuss with the instructor to design the fields in the CRD and modify the logic of the original backend code to manage the new CRD
   - Investigate the front-end and back-end communication methods of S2I

**Reply by Ganbingkun** [reply]
# weekly report 3
- Project: KubeSphere-S2I
- Reporter: Bingkun Gan
- Progress:
   - Based on Shipwright Yaml and using Buildpacks, compiled some basic images and source code into new images.
   - Study Client-go and prepare for adjusting backend code.
- Planning:
   - Keep using Shipwright and Buildpacks to compile other language basic images and source code into new images.
   - Begin use Kubebuilder ,client-go and go-restful to adjust backend code.

**Reply by Ganbingkun** [reply]
# weekly report 4
- Project: KubeSphere-S2I
- Reporter: Bingkun Gan
- Problem：
   - Some images produced using open-source tools cannot be used by Shipwright. I am currently looking for alternative tools
- Progress:
   - Clarify the backend code logic and proceed with writing the backend code
- Planning:
   - Complete the backend code over the next two weeks, along with comprehending the frontend React code.

**Reply by Ganbingkun** [reply]
# weekly report 4
- Project: KubeSphere-S2I
- Reporter: Bingkun Gan
- Progress:
   - Complete the backend code
   - Complete the backend code test case
- Planning:
   - Continue testing the backend code
   - Rewrite the front-end code

**Reply by Ganbingkun** [reply]
# weekly report 6
- Project: KubeSphere-S2I
- Reporter: Bingkun Gan
   - Test the backend
   - Submit an installation package PR

**Comment by cherryFloris** [comment]
### Weekly Meeting Report

- **Prerequisite**:

  -   kubebuilder learning
  -   service/kube-proxy learning
 
- **Project**：OpenELB
- **Reporter**: Chenying Guo
- **Progress**: 
    -   Set up the development environment. 
    -   Combed through openelb's `ipam logic`. 
    -   Complete the specified `eip allocation` and the specified `IP allocation` function, this part of the functionality is still being. tested.
- **Problem**: Part of the code is a bit of a detour, I have to find the instructor to understand clearly.
- **Plan**：Modify the existing default `eip logic` and improve manual protocol assignment.

**Reply by cherryFloris** [reply]
### Weekly Meeting Report  2
- **Project**：OpenELB
- **Reporter**: Chenying Guo
- **Progress**: 
    -   Recalibrate the scheme, clarify the `Controller` and `Speaker` of the division of labor, and do not achieve the specified protocol `IP allocation`
    -  [ Delete the webhook of the default eip](https://github.com/openelb/openelb/pull/348). 
    -   The `namespace `fields related to eip resources are being adjusted.
- **Problem**: The scheme was readjusted.
- **Plan**：
    - 	Improve the default `eip logic` after the modification.
    - 	Develop the binding logic to the `namespace`.

**Reply by cherryFloris** [reply]
### Weekly Meeting Report 3
- **Project**：OpenELB
- **Reporter**: Chenying Guo
- **Progress**: [allocate eip to specified namespace #352](https://github.com/openelb/openelb/pull/352)
    -   Deleted protocol-related verification because of the assignment of the specified protocol is not supported. 
    -   Improved default eip function since [the previous pr](https://github.com/openelb/openelb/pull/348) removed the default `eip`. 
    -   Updated the eip resource, adding some fields bound to the `namespace`.
    -   Completed the development of the `eip` binding `namespace` function, but the `Speaker` also needs to make some corresponding improvements.
- **Plan**：
    - 	Perfect the `Speaker`.
    - 	Add tests and fix problems.

**Reply by cherryFloris** [reply]
### Weekly Meeting Report 4
- **Project**：OpenELB
- **Reporter**: Chenying Guo
- **Progress**: 
    -   [Improved `speaker` functions.](https://github.com/openelb/openelb/pull/352)
    -   Fixed the problem.
- **Plan**：
    - 	Test in depth and fix the problem if found.
    - 	Replenish use cases + documentation.

**Reply by cherryFloris** [reply]
### Weekly Meeting Report 5
- **Project**：OpenELB
- **Reporter**: Chenying Guo
- **Progress**: 
    -   Replenish use cases.
- **Plan**：
    - 	Replenish use cases + documentation.

**Reply by cherryFloris** [reply]
### Weekly Meeting Report 6
- **Project**：OpenELB
- **Reporter**: Chenying Guo
- **Progress**: 
    -   Finish [use cases](https://github.com/openelb/openelb/pull/357) + [documentation](https://github.com/openelb/website/pull/86).

**Comment by B1F030** [comment]
### Weekly Meeting Report-1

- **Project**: OpenFunction Trigger
- **Reporter**: Guanjing Zhang
- **Progress**:
  - read OpenFunction source code(v1beta2/serving-types)
  - deployed ofn local environment and finished quick start examples
  - trying to verify feasibility of dapr & keda framework
  - learning the way of injecting dapr sidecar

**Reply by B1F030** [reply]
### Weekly Meeting Report-2

- **Project**: OpenFunction Trigger
- **Reporter**: Guanjing Zhang
- **Progress**:
  - verified feasibility of dapr & keda framework by sample demonstration
  - proposal completed and ready to commit
  - add an `HTTPScaledObject` under `scaleOptions.keda`
  - based on CRD design, analysing reconcile code

**Reply by B1F030** [reply]
### Weekly Meeting Report-3

- **Project**: OpenFunction Trigger
- **Reporter**: Guanjing Zhang
- **Progress**:
  - proposal merged
  - serving runtime based on keda http-addon ready to test
  - keda version need to be updated from v2.8.1 to v2.10.1, autoscaling need to be updated from v2beta2 to v2

**Reply by B1F030** [reply]
### Weekly Meeting Report-4

- **Project**: OpenFunction Trigger
- **Reporter**: Guanjing Zhang
- **Progress**:
  - PR ready to merge
  - relative documents need to be updated
  - project final report in progress

**Reply by B1F030** [reply]
### Weekly Meeting Report-5

- **Project**: OpenFunction Trigger
- **Reporter**: Guanjing Zhang
- **Progress**:
  - replace ingress with gateway

**Reply by B1F030** [reply]
### Weekly Meeting Report-6

- **Project**: OpenFunction Trigger
- **Reporter**: Guanjing Zhang
- **Progress**:
  - PR merged, project finished

**Comment by sjliu1** [comment]
WWeekly Meeting Report

> Project: Fluent Operator Collector 
> Reporter: Shuaijun Liu

_1st report_

>  Progress:
- learn docker, kubernetes related content, and installed the environment;
- investigate input components and learn how to install components;
- read the Fluent Bit official documentation to learn how Fluent Bit logs are collected.

_2nd report_

>  Progress:
- Continued to learn k8s;
- Installed the single-node cluster of kubesphere, and learned various components of kubernetes, such as apiserver, etcd and - controller-manager;
- Installed fluent-operator using helm.
>  Plan:
- Investigate input components and build a platform for self-testing.


_3rd report_

>  Progress:
- Based on the Fluentbit document, the code part of the two input components, `collectd `and `forward`, has been integrated with the Fluent-operator.
- `opentelemetry`, `http` and `mqtt` parts are currently in progress.
- Built the Kubernetes platform and started testing `collectd` and `statsd`.
>  Plan:
-  Finish the rest of `opentelemetry`, `http`, `mqtt`.
-  Continue testing the completed components.


_4th report_
>  Progress:
- Complete the code of components `opentelemetry`, `http` and `mqtt` 
- Test components such as `collectd` and `statsd`
>  Plan:
- Complete the code parts of the remaining components and test them
- Organize relevant codes and prepare for submission


_5th report_
>  Progress:
- Submitted codes for components `opentelemetry`, `http` and `mqtt` 
- Continue testing the components `collectd` and `statsd`
>  Plan:
- Submit codes for the components `collectd` and `statsd`

_6th report_
>  Progress:
- Complete code merge
- Write project completion documents
